IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -121,38 +171,49 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json
                                  error:(NSError * __autoreleasing *)error
 {
     MSIDMacCredentialStorageItem *instance = [self init];
-    
+
     if (instance)
     {
-        for (NSString *credentialKey in json)
+        for (NSString *typeKey in json)
         {
-            NSDictionary *credentialDict = [json msidObjectForKey:credentialKey ofClass:[NSDictionary class]];
+            NSMutableDictionary *typeDict = [NSMutableDictionary dictionary];
+            NSDictionary *subDict = [json msidObjectForKey:typeKey ofClass:[NSDictionary class]];
             
-            if (credentialDict)
+            if (subDict)
             {
-                MSIDCredentialCacheItem *credential = [[MSIDCredentialCacheItem alloc] initWithJSONDictionary:credentialDict error:error];
-                
-                if (credential)
+                for (NSString *itemKey in subDict)
                 {
-                    MSIDCacheKey *storedCredentialKey = [credential createCredentialCacheKey];
+                    NSDictionary *itemDict = [subDict msidObjectForKey:itemKey ofClass:[NSDictionary class]];
                     
-                    if (storedCredentialKey)
+                    if (itemDict)
                     {
-                        [instance.cacheObjects setObject:credential forKey:storedCredentialKey];
+                        id<MSIDJsonSerializable, MSIDKeyGenerator> storedItem = [self getItemTypeFromString:itemDict forKey:typeKey error:error];
```

Comment:
nit: rename to getItemWithType? Otherwise reads like you're retrieving item type and not an item
Also, do we need to check the protocol? Can we make it a responsibility of the getItemWithType method to return item that conforms to a correct protocol and do the check if needed? if it doesn't, it can return nil and the calling method can do a nil-check only.

Reply:
Agree. Updated the api name to getItemWithType and it already takes care of returning the correct object that conforms to the required protocol.